mcmcfit.py

```diff
@@ -338,37 +338,36 @@ def createGP(self,phi):
         tau = np.exp(ln_tau.currVal)
 
         # Calculate kernels for both out of and in eclipse WD eclipse
-        # Kernel inside of WD has smaller amplitude than that of outside eclipse,
-        #k_in  = ampin*GP.ExpSquaredKernel(tau)
-        #k_out  = ampout*GP.ExpSquaredKernel(tau)
-        k_in  = ampin*GP.Matern32Kernel(tau)
-        k_out = ampout*GP.Matern32Kernel(tau)
-        #k_in  = ampin*GP.ExpKernel(tau)
-        #k_out = ampout*GP.ExpKernel(tau)
-
+        # Kernel inside of WD has smaller amplitude than that of outside eclipse
+        # First, get the changepoints
         changepoints = self.calcChangepoints(phi)
 
-        # Depending on number of changepoints, create kernel structure
-        kernel_struc = [k_out]
-        for k in range (int( phi.min() ), int( phi.max() )+1, 1):
-            kernel_struc.append(k_in)
-            kernel_struc.append(k_out)
-
-        # Create kernel with changepoints
-        kernel = GP.DrasticChangepointKernel(kernel_struc,changepoints)
-
-        '''k1 = GP.Matern32Kernel(tau)
-
-        gp_pars = np.array([ampout,ampin,ampout])
-        changepoints = self.calcChangepoints(phi)
+        # We need to make a fairly complex kernel.
+        # Initialise with an 'out of eclipse' kernel, that runs from negative infinity to the first changepoint.
+        kernel = ampout * g.kernels.Matern32Kernel(tau, block=(-np.inf, changepoints[0]) )
```

Comment:
Yeah - this is quite hard to get right. I think you've done it here, but we could manage it with fewer kernels, which might speed things up. It means changing the meaning of the parameters, so ```ampout``` becomes the *extra* amplitude visible when the WD is visible. Something like the figure below:

![Paper Ideas 2019 04 30](https://user-images.githubusercontent.com/4570807/56971082-770ec580-6b60-11e9-8505-988d2418d83f.png)

I think that could use simpler code and would use 5 kernels, whereas your code needs 7...